controller/replica_controller.go

```diff
@@ -625,6 +643,35 @@ func (rc *ReplicaController) DeleteInstance(obj interface{}) (err error) {
 	return nil
 }
 
+func (rc *ReplicaController) ReplicaRemove(replica *longhorn.Replica) error {
+	engine, err := rc.ds.GetEngine(replica.Spec.EngineName)
+	if err != nil {
+		return err
+	}
+
+	addr, ok := engine.Status.CurrentReplicaAddressMap[replica.Name]
+	if !ok {
+		rc.logger.Warnf("replica %s is not found in engine %s replica address map", replica.Name, engine.Name)
+		return nil
+	}
+
+	engineClientProxy, err := rc.getEngineClientProxy(engine, engine.Status.CurrentImage)
+	if err != nil {
+		return err
+	}
```

Comment:
Should close the connection
```suggestion
	engineClientProxy, err := rc.getEngineClientProxy(engine, engine.Status.CurrentImage)
	if err != nil {
		return errors.Wrapf(err, "failed to get the engine client proxy for ...)
	}
	defer engineClientProxy.Close()
```